File: YLDropMenuTest/YLDropMenuTest/ViewController.m
//
//  ViewController.m
//  YLDropMenuTest
//
//  Created by fanyl on 2019/9/9.
//  Copyright © 2019 fanyl. All rights reserved.
//

#import "ViewController.h"

#import "ViewTestVC.h"
#import "ViewControllerTestVC.h"

#import <Masonry/Masonry.h>

@interface ViewController () <UITableViewDelegate, UITableViewDataSource>
{
    ViewControllerTestVC *_vc2;
}
@property (nonatomic, strong) UITableView *tableView;
@property (nonatomic, strong) NSArray<NSDictionary<NSString *, UIViewController *> *> *dataSource;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = @"Demo";
    
    [self.view addSubview:self.tableView];
    
    [self.tableView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.edges.equalTo(self.view).insets(UIEdgeInsetsZero);
    }];
    
    _vc2 = [[ViewControllerTestVC alloc] initWithNibName:nil bundle:nil];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    ViewTestVC *vc = [[ViewTestVC alloc] initWithNibName:nil bundle:nil];
    

    self.dataSource = @[@{@"View 形式调用" : vc}, @{@"ViewController 形式调用" : _vc2}];
    
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataSource.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:NSStringFromClass(UITableViewCell.class) forIndexPath:indexPath];
    cell.textLabel.text = self.dataSource[indexPath.row].allKeys.firstObject;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
    NSDictionary<NSString *, UIViewController *> *dic = (NSDictionary *)self.dataSource[indexPath.row];
    [self.navigationController pushViewController:dic.allValues.firstObject animated:YES];
}

- (UITableView *)tableView {
    if (!_tableView) {
        UITableView *tmp = [[UITableView alloc] initWithFrame:CGRectZero style:UITableViewStylePlain];
        [tmp registerClass:UITableViewCell.class forCellReuseIdentifier:NSStringFromClass(UITableViewCell.class)];
        tmp.dataSource = self;
        tmp.delegate = self;
        _tableView = tmp;
    }
    return _tableView;
}

@end

File: my_question/MyQuestion/MyQuestion/ViewController.swift
//
//  ViewController.swift
//  MyQuestion
//
//  Created by oker on 2024/5/11.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    private lazy var tableview: UITableView = {
        var tmp = UITableView(frame: .zero, style: .plain)
        tmp.delegate = self
        tmp.dataSource = self
        tmp.register(UITableViewCell.self, forCellReuseIdentifier: "id")
        return tmp
    }()
    
    private let dataSource = ["文字+图片样式，可折行，图片左贴文字", "UITableView 在顶部加入View，利用偏移", "顶部视图包含左右两个视图，顶部视图的高度取决于左右视图的大者"]

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        view.addSubview(tableview)
        tableview.snp.makeConstraints { make in
            make.edges.equalTo(view)
        }
    }

}

extension ViewController {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataSource.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "id", for: indexPath)
        cell.textLabel?.text = dataSource[indexPath.row]
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        if indexPath.row == 0 {
            let vc = LabelImageVC.init()
            self.navigationController?.pushViewController(vc, animated: false)
        }
        if indexPath.row == 1 {
            let vc = TopViewTableViewVC.init()
            self.navigationController?.pushViewController(vc, animated: false)
        }
        if indexPath.row == 2 {
            let vc = TopViewVC.init()
            self.navigationController?.pushViewController(vc, animated: false)
        }
    }
}

